dotcom-rendering/src/model/enhanceTableOfContents.ts

```diff
@@ -0,0 +1,122 @@
+import { JSDOM } from 'jsdom';
+
+const isHeading = (element: CAPIElement, level: 'H2' | 'H3' | 'H4') => {
+	if (
```

Comment:
Why do we need to check for these types here when we type them in each of the `isH` functions below? Is this just an extra layer of security?

Reply:
I played around with these checks to try and avoid bloating things but this if statement is needed it seems.

Because `element` can be any type of element, it needs the code to narrow things down. If I change the type of `element` to be `SubheadingBlockElement | TextBlockElement` then the calling function has a problem because that is passing in CAPIElement, which could be anything